src/api/GetProvider.ts

```diff
@@ -0,0 +1,7 @@
+import { ConfigurationService, GetProviderRequest } from '.';
+
+/**
+ * Returns the class of a ConfigurationService based on the configurationType
+ * @param getProviderRequest
+ */
+export type GetProvider = (getProviderRequest: GetProviderRequest) => new (configurationServiceConfig: any) => ConfigurationService;
```

Comment:
I'm not sure to understand your syntax here. 
Why not `type GetProvider = (getProviderRequest: GetProviderRequest) => ConfigurationService;` ?